worlds/ahit/Regions.py

```diff
@@ -0,0 +1,946 @@
+    entrance_name = act_entrances.get(region)
+    if entrance_name != "":
+        is_first_act: bool = act_chapters.get(region) == get_first_chapter_region(world).name \
+                             and ("Act 1" in entrance_name or "Free Roam" in entrance_name)
+
+        if is_first_act and not is_valid_first_act(world, world.multiworld.get_region(act, world.player)):
+            return False
+
+    # Don't allow straight up impossible mappings
+    if (region == "Time Rift - Curly Tail Trail"
+       or region == "Time Rift - The Twilight Bell"
+       or region == "The Illness has Spread") \
+       and act == "Alpine Free Roam":
+        return False
+
+    if (region == "Rush Hour" or region == "Time Rift - Rumbi Factory") and act == "Nyakuza Free Roam":
+        return False
+
+    if region == "Time Rift - The Owl Express" and act == "Murder on the Owl Express":
+        return False
+
+    if region == "Time Rift - Deep Sea" and act == "Bon Voyage!":
+        return False
+
+    return any(a.name == world.options.ActPlando.get(region) for a in world.multiworld.get_regions(world.player))
+
+
+def create_region(world: "HatInTimeWorld", name: str) -> Region:
+    reg = Region(name, world.player, world.multiworld)
+
+    for (key, data) in location_table.items():
+        if world.is_dw_only():
+            break
+
+        if data.nyakuza_thug != "":
+            continue
+
+        if data.region == name:
+            if key in storybook_pages.keys() \
+               and world.options.ShuffleStorybookPages.value == 0:
+                continue
+
+            location = HatInTimeLocation(world.player, key, data.id, reg)
+            reg.locations.append(location)
+            if location.name in shop_locations:
+                world.shop_locs.append(location.name)
+
+    world.multiworld.regions.append(reg)
+    return reg
+
+
+def create_badge_seller(world: "HatInTimeWorld") -> Region:
+    badge_seller = Region("Badge Seller", world.player, world.multiworld)
+    world.multiworld.regions.append(badge_seller)
+    count = 0
+    max_items = 0
+
+    if world.options.BadgeSellerMaxItems.value > 0:
+        max_items = world.random.randint(world.options.BadgeSellerMinItems.value,
+                                         world.options.BadgeSellerMaxItems.value)
+
+    if max_items <= 0:
+        world.badge_seller_count = 0
+        return badge_seller
+
+    for (key, data) in shop_locations.items():
+        if "Badge Seller" not in key:
+            continue
+
+        location = HatInTimeLocation(world.player, key, data.id, badge_seller)
+        badge_seller.locations.append(location)
+        world.shop_locs.append(location.name)
+
+        count += 1
+        if count >= max_items:
+            break
+
+    world.badge_seller_count = max_items
+    return badge_seller
+
+
+def connect_regions(start_region: Region, exit_region: Region, entrancename: str, player: int) -> Entrance:
```

Comment:
this function is superceded by `Region.connect`. i.e. `start_region.connect(exit_region, entrancename)`

Reply:
done